config.py

```diff
@@ -302,12 +310,21 @@ def get_config(parse_args=True, cfg_path=None, options=None):
         path = os.path.dirname(path)
 
         config_path = get_config_path(cfg_path, os_name=get_os())
-        config = ConfigParser.ConfigParser()
+        if options is not None and options.enable_env:
+            log.info("Interpolation of environment variables in config is enabled")
+            config = ConfigParser.ConfigParser(os.environ)
```

Comment:
@delitescere @LeoCavaille 
You may want to add `ConfigParser.optionxform = str` before `config = ConfigParser.ConfigParser(os.environ)` so that the keys in `os.environ` are treated as case sensitive.  

For example:
```python
>>> from ConfigParser import ConfigParser
>>> import os
>>> os.environ['XY'] = 'foo'
>>> os.environ['Xy'] = 'bar'
>>> os.environ['xY'] = 'baz'
>>> print '\n'.join(sorted([':'.join((k, v)) for k, v in os.environ.items() if k.lower() == 'xy']))
XY:foo
Xy:baz
xY:bar

>>> cfgp = ConfigParser(os.environ)
>>> print '\n'.join(sorted([':'.join((k, v)) for (k, v) in cfgp.defaults().items() if k.lower() == 'xy']))
xy:foo

>>> ConfigParser.optionxform = str
>>> cfgp_cs = ConfigParser(os.environ)
>>> print '\n'.join(sorted([':'.join((k, v)) for (k, v) in cfgp_cs.defaults().items() if k.lower() == 'xy']))
XY:foo
Xy:baz
xY:bar

>>> assert([(k, v) for (k, v) in cfgp_cs.defaults().items() if k.lower() == 'xy'] == [(k, v) for (k, v) in os.environ.items() if k.lower() == 'xy'])
>>> assert([(k, v) for (k, v) in cfgp.defaults().items() if k.lower() == 'xy'] == [(k, v) for (k, v) in os.environ.items() if k.lower() == 'xy'])
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-43-1f97baeb1e97> in <module>()
----> 1 assert([(k, v) for (k, v) in cfgp.defaults().items() if k.lower() == 'xy'] == [(k, v) for (k, v) in os.environ.items() if k.lower() == 'xy'])

AssertionError:
```